### ISSUE TITLE
Direction, Align, Position and ViewScroll not working for ControlledMenu with anchorPoint

### ISSUE THREAD

**Issue by panciumihai** [szhsin/react-menu#694]
React/React-dom version: 17
React-Menu version: ^3.1.2

## Describe the bug
Hello @szhsin. I tried to use ControlledMenu with anchorPoint and at the same time I wanted to set the following props: direction, align, position, offsetX, offsetY. All of them are not working if you are using anchorPoint as positioning strategy. I tried also with a fresh project to confirm.

## To Reproduce
This is an example of how I wanted to use it:
```js
export default function Example() {
  const [menuProps, toggleMenu] = useMenuState();
  const [anchorPoint, setAnchorPoint] = useState({ x: 0, y: 0 });
  const ref = useRef();

  return (
    <div
      ref={ref}
      style={{
        background: 'black',
        height: '300px',
        width: '400px',
        marginLeft: '300px',
        marginBottom: '300px',
      }}
      onContextMenu={(e) => {
        e.preventDefault();
        setAnchorPoint({ x: e.clientX, y: e.clientY });
        toggleMenu(true);
      }}
    >
      Right click to open context menu
      <ControlledMenu
        {...menuProps}
        // anchorRef={ref}
        anchorPoint={anchorPoint}
        direction='top'
        align='center'
        offsetX={50}
        onClose={() => toggleMenu(false)}
      >
        <MenuItem>Cut</MenuItem>
        <MenuItem>Copy</MenuItem>
        <MenuItem>Paste</MenuItem>
      </ControlledMenu>
    </div>
  );
}
```

It looks like this, whatever props are set (red dot is the anchorPoint location):
![image](https://user-images.githubusercontent.com/22646067/197764967-a219d4dc-cb20-493a-a4ab-b327a5c5e60a.png)

## Expected behavior
For the giving props in the above example, the expected behavior should be the following:
![image](https://user-images.githubusercontent.com/22646067/197765282-68ab1420-3022-445b-a597-9b96e86886b6.png)

If this is not a bug, then it will be a nice feature.
Thank you and let me know if I can help with more info!

**Comment by panciumihai** [comment]
Hello,

I managed to implement the above behavior. Here is the PR: https://github.com/szhsin/react-menu/pull/699
Now it looks like this:
![image](https://user-images.githubusercontent.com/22646067/197963396-54368da1-d5b5-4bb5-a1d1-616d93517824.png)

For the following configuration:
```js
      <ControlledMenu
        {...menuProps}
        anchorPoint={anchorPoint}
        direction={'top'}
        align={'center'}
        offsetX={-20}
        offsetY={10}
        onClose={() => toggleMenu(false)}
      >
        <MenuItem>Cut</MenuItem>
        <MenuItem>Copy</MenuItem>
        <MenuItem>Paste</MenuItem>
      </ControlledMenu>
```


I tested it and unit tests are passing.
![image](https://user-images.githubusercontent.com/22646067/197959724-bdd14897-9635-43cd-afe9-ff0bf5e55a99.png)

**Comment by szhsin** [comment]
Hi @panciumihai , 

The context menu currently ignores those props and doesn't take them into account for calculating position, but it could be a good feature if they are supported for context menu.

Also, thanks for the PR and we will check it out and assess the change.

**Comment by szhsin** [comment]
@panciumihai, your PR has been merged and the enhancement is available in v3.2.1
